cmd/kubeadm/app/util/version.go

```diff
@@ -162,30 +168,45 @@ func splitVersion(version string) (string, string, error) {
 	default:
 		urlSuffix = "release"
 	}
-	url := fmt.Sprintf("%s/%s", kubeReleaseBucketURL, urlSuffix)
-	return url, subs[0][3], nil
+	baseURL := fmt.Sprintf("%s/%s", kubeReleaseBucketURL, urlSuffix)
+	return baseURL, subs[0][3], nil
 }
 
-// Internal helper: return content of URL
-func fetchFromURL(url string, timeout time.Duration) (string, error) {
-	klog.V(2).Infof("fetching Kubernetes version from URL: %s", url)
-	client := &http.Client{Timeout: timeout, Transport: netutil.SetOldTransportDefaults(&http.Transport{})}
-	resp, err := client.Get(url)
-	if err != nil {
-		return "", errors.Errorf("unable to get URL %q: %s", url, err.Error())
-	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return "", errors.Errorf("unable to read content of URL %q: %s", url, err.Error())
-	}
-	bodyString := strings.TrimSpace(string(body))
+// resolveVersionLabel resolves version label into normalized version string
+func resolveVersionLabel(bucketURL, versionLabel string) (string, error) {
+	client := &http.Client{Timeout: getReleaseVersionTimeout, Transport: netutil.SetOldTransportDefaults(&http.Transport{})}
+	ver := versionLabel
```

Comment:
Since you are passing `versionLabel` to this function by value, I don't think you need to assign it to another variable.

Reply:
Thanks! Fixed.